src/trunk.c

```diff
@@ -7905,68 +7968,109 @@ trunk_print_locked_node(platform_log_handle *log_handle,
-   // clang-format on
+}
+
+/*
+ * trunk_print_branches_and_bundles() --
+ *
+ * Iterate through arrays of bundles and sub-bundles on a trunk page.
+ * Print contents of those structures.
+ */
+static void
+trunk_print_branches_and_bundles(platform_log_handle *log_handle,
+                                 trunk_handle        *spl,
+                                 page_handle         *node)
+{
    uint16 start_branch = trunk_start_branch(spl, node);
    uint16 end_branch   = trunk_end_branch(spl, node);
    uint16 start_bundle = trunk_start_bundle(spl, node);
    uint16 end_bundle   = trunk_end_bundle(spl, node);
    uint16 start_sb     = trunk_start_subbundle(spl, node);
    uint16 end_sb       = trunk_end_subbundle(spl, node);
+
+   // clang-format off
+   platform_log(log_handle, "|--------------------------------------------------------------------------------------------------|\n");
+   platform_log(log_handle, "|                              BRANCHES AND [SUB]BUNDLES                                           |\n");
+   platform_log(log_handle, "|start_branch=%-2u end_branch=%-2u start_bundle=%-2u end_bundle=%-2u start_sb=%-2u end_sb=%-2u%-17s|\n",
+                start_branch,
+                end_branch,
+                start_bundle,
+                end_bundle,
+                start_sb,
+                end_sb,
+                " ");
+   platform_log(log_handle, "|--------------------------------------------------------------------------------------------------|\n");
+   platform_log(log_handle, "|   # |          point addr         | filter1 addr | filter2 addr | filter3 addr |                 |\n");
+   platform_log(log_handle, "|     |    pivot/bundle/subbundle   |  num tuples  |              |              |                 |\n");
+   platform_log(log_handle, "|-----|--------------|--------------|--------------|--------------|--------------|-----------------|\n");
+   // clang-format on
+
+   // Iterate through all the branches ...
    for (uint16 branch_no = start_branch; branch_no != end_branch;
         branch_no        = trunk_add_branch_number(spl, branch_no, 1))
    {
+      // Generate marker line if current branch is a pivot's start branch
       for (uint16 pivot_no = 0; pivot_no < trunk_num_children(spl, node);
            pivot_no++) {
          if (branch_no == trunk_pivot_start_branch(spl, node, pivot_no)) {
             // clang-format off
-            platform_log(log_handle, "|     |        -- pivot %2u --       |              |              |              |                |\n",
-                                pivot_no);
+            platform_log(log_handle, "|     |        -- pivot %2u --       |              |              |              |                 |\n",
+                         pivot_no);
             // clang-format on
          }
       }
+
+      // Iterate through all the bundles ...
```

Comment:
This comment is confusing, since this is a nested iteration, and all we're really doing is searching for bundles that start at this branch.

Reply:
Deleted existing comment, and reworded as you described.